qiskit/providers/backend_compat.py

```diff
@@ -0,0 +1,315 @@
+    defaults: PulseDefaults = None,
+    custom_name_mapping: Optional[Dict[str, "Operation"]] = None,
+) -> Target:
+    """Uses configuration, properties and pulse defaults
+    to construct and return Target class.
+    """
+    # Standard gates library mapping, multicontrolled gates not included since they're
+    # variable width
+    name_mapping = {
+        "id": IGate(),
+        "sx": SXGate(),
+        "x": XGate(),
+        "cx": CXGate(),
+        "rz": RZGate(Parameter("λ")),
+        "r": RGate(Parameter("ϴ"), Parameter("φ")),
+        "reset": Reset(),
+        "c3sx": C3SXGate(),
+        "ccx": CCXGate(),
+        "dcx": DCXGate(),
+        "ch": CHGate(),
+        "cp": CPhaseGate(Parameter("ϴ")),
+        "crx": CRXGate(Parameter("ϴ")),
+        "cry": CRYGate(Parameter("ϴ")),
+        "crz": CRZGate(Parameter("ϴ")),
+        "cswap": CSwapGate(),
+        "csx": CSXGate(),
+        "cu": CUGate(Parameter("ϴ"), Parameter("φ"), Parameter("λ")),
+        "cu1": CU1Gate(Parameter("λ")),
+        "cu3": CU3Gate(Parameter("ϴ"), Parameter("φ"), Parameter("λ")),
+        "cy": CYGate(),
+        "cz": CZGate(),
+        "ccz": CCZGate(),
+        "h": HGate(),
+        "p": PhaseGate(Parameter("ϴ")),
+        "rccx": RCCXGate(),
+        "rcccx": RC3XGate(),
+        "rx": RXGate(Parameter("ϴ")),
+        "rxx": RXXGate(Parameter("ϴ")),
+        "ry": RYGate(Parameter("ϴ")),
+        "ryy": RYYGate(Parameter("ϴ")),
+        "rzz": RZZGate(Parameter("ϴ")),
+        "rzx": RZXGate(Parameter("ϴ")),
+        "xx_minus_yy": XXMinusYYGate(Parameter("ϴ")),
+        "xx_plus_yy": XXPlusYYGate(Parameter("ϴ")),
+        "ecr": ECRGate(),
+        "s": SGate(),
+        "sdg": SdgGate(),
+        "cs": CSGate(),
+        "csdg": CSdgGate(),
+        "swap": SwapGate(),
+        "iswap": iSwapGate(),
+        "sxdg": SXdgGate(),
+        "t": TGate(),
+        "tdg": TdgGate(),
+        "u": UGate(Parameter("ϴ"), Parameter("φ"), Parameter("λ")),
+        "u1": U1Gate(Parameter("λ")),
+        "u2": U2Gate(Parameter("φ"), Parameter("λ")),
+        "u3": U3Gate(Parameter("ϴ"), Parameter("φ"), Parameter("λ")),
+        "y": YGate(),
+        "z": ZGate(),
+        "delay": Delay(),
+    }
+    custom_gates = {}
+    target = None
+    if custom_name_mapping is not None:
+        name_mapping.update(custom_name_mapping)
+    # Parse from properties if it exsits
+    if properties is not None:
+        qubit_properties = qubit_props_list_from_props(properties=properties)
+        target = Target(num_qubits=configuration.n_qubits, qubit_properties=qubit_properties)
+        # Parse instructions
+        gates: Dict[str, Any] = {}
+        for gate in properties.gates:
+            name = gate.gate
+            if name in name_mapping:
+                if name not in gates:
+                    gates[name] = {}
+            elif name not in custom_gates:
+                custom_gate = Gate(name, len(gate.qubits), [])
```

Comment:
Maybe this could issue a warning for possibility of missing parameter? I a user doesn't notice custom mapping, I guess one will get confusing random error message from somewhere in the stack.

Reply:
Oh, I should change this to a raise, I meant to do that before it got lost in my response to the earlier comments. I don't think we ever want to do this, if we can't map the name to a gate this will just lead to broken behavior.

Reply:
Made a raise in: https://github.com/Qiskit/qiskit-terra/pull/8759/commits/86434cd9cf5cd7e95b0a14951f54996550bf1552